pkg/core/mpt/trie.go

```diff
@@ -348,3 +348,33 @@ func (t *Trie) getFromStore(h util.Uint256) (Node, error) {
 	}
 	return n.Node, nil
 }
+
+// Collapse compresses all nodes at depth n to the hash nodes.
```

Comment:
It's only safe to do this after `Flush`, so it should either be invoked internally (for collapsed elements?) or it should be mentioned here. Probably a combined method would be more appropriate.

Reply:
This is a separate function, because we _must_ perform `Flush` every block, however `Collapse` can be called once per N blocks if needed. Not a big deal now probably, but can be useful e.g. during restore.

Reply:
What about a `bool` flag to `Collapse` specifying if everything should be flushed?